packages/variable-editor/src/components/variables/master/VariablesMasterContent.tsx

```diff
@@ -25,9 +26,15 @@ import { ValidationRow } from './ValidationRow';
 import './VariablesMasterContent.css';
 
 export const VariablesMasterContent = () => {
-  const { variables, setVariables, setSelectedVariable } = useAppContext();
+  const { variables, setVariables, setSelectedVariable, detail, setDetail } = useAppContext();
 
-  const selection = useTableSelect<Variable>();
+  const selection = useTableSelect<Variable>({
+    onSelect: selectedRows => {
+      const selectedRowId = Object.keys(selectedRows).find(key => selectedRows[key]);
+      const selectedVariable = table.getRowModel().flatRows.find(row => row.id === selectedRowId)?.id;
+      setSelectedVariable(selectedVariable ? toTreePath(selectedVariable) : []);
```

Comment:
You could make `toTreePath` accept `undefined` as an argument and then simply change this line to:

```suggestion
      setSelectedVariable(toTreePath(selectedVariable));
```

If `selectedVariable` is `undefined`, `toTreePath` should then return an empty Array (should already be the case).

Reply:
nice, thanks 👍 had to add a undefined in the toTreePath in order to pass a undefined selectedVariable ✌️